test/evp_test.c

```diff
@@ -168,6 +168,139 @@ static int test_bin(const char *value, unsigned char **buf, size_t *buflen)
     *buflen = len;
     return 1;
 }
+
+/*
+ * Structure used to hold a list of blocks of memory to test
+ * calls to "update" like functions.
+ */
+
+typedef struct {
+    unsigned char *buf;
+    size_t buflen;
+    size_t count;
+    int count_set;
+} EVP_TEST_BUFFER;
+
+DEFINE_STACK_OF(EVP_TEST_BUFFER)
+
+static void evp_test_buffer_free(EVP_TEST_BUFFER *db)
+{
+    if (db != NULL) {
+        OPENSSL_free(db->buf);
+        OPENSSL_free(db);
+    }
+}
+
+/* append buffer to a list */
+
+static int evp_test_buffer_append(const char *value,
+                                  STACK_OF(EVP_TEST_BUFFER) **sk)
+{
+    EVP_TEST_BUFFER *db = NULL;
+
+    db = OPENSSL_malloc(sizeof(*db));
+    if (db == NULL)
+        goto err;
+
+    if (!test_bin(value, &db->buf, &db->buflen))
+        goto err;
+    db->count = 1;
+    db->count_set = 0;
+
+    if (*sk == NULL) {
+        *sk = sk_EVP_TEST_BUFFER_new_null();
+        if (*sk == NULL)
+            goto err;
+    }
+
+    if (!sk_EVP_TEST_BUFFER_push(*sk, db))
+        goto err;
+
+    return 1;
+
+    err:
+    evp_test_buffer_free(db);
+
+    return 0;
+}
+
+/* replace last buffer in list with copies of itself */
+
```

Comment:
delete blank line between comment and function.
maybe use the multi-line comment style for function comments?  optional.